modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectActionLocalServiceImpl.java

```diff
@@ -411,6 +417,30 @@ private void _validateParametersUnicodeProperties(
 				_objectDefinitionPersistence.fetchByPrimaryKey(
 					objectDefinitionId);
 
+			if (parametersUnicodeProperties.containsKey(
+					"objectDefinitionExternalReferenceCode")) {
+
+				String objectDefinitionExternalReferenceCode =
+					GetterUtil.getString(
+						parametersUnicodeProperties.remove(
+							"objectDefinitionExternalReferenceCode"));
+
+				ObjectDefinition existingObjectDefinition =
+					_objectDefinitionPersistence.fetchByERC_C(
+						objectDefinitionExternalReferenceCode, companyId);
+
+				if (!Objects.isNull(existingObjectDefinition)) {
+					objectDefinition = existingObjectDefinition;
+
+					objectDefinitionId =
+						objectDefinition.getObjectDefinitionId();
+
+					parametersUnicodeProperties.put(
+						"objectDefinitionId",
+						String.valueOf(objectDefinitionId));
```

Comment:
Could you inline the variable "objectDefinitionId",? Because you are only using once.